Framework/PythonInterface/mantid/_testing/AssertAlmostEqual.py

```diff
@@ -27,26 +30,66 @@ def assert_almost_equal(Workspace1, Workspace2, rtol=Property.EMPTY_DBL, atol=Pr
     AssertionError
         If Workspace1 and Workspace2 are not equal up to specified precision
     ValueError
-        If atol and rtol are both not provided
+        If atol or rtol are below zero
+    ValueError
+        If kwargs contains keys that cshould instead be set by atol or rtol
 
     """
     # check arguments
-    if len(set(kwargs.keys()).intersection({"Workspace1", "Workspace2", "Tolerance", "ToleranceRelErr"})):
-        raise ValueError("Workspace1, Workspace2, Tolerance, ToleranceRelErr cannot be passed as additional parameters")
+    if len(set(kwargs.keys()).intersection({"Tolerance", "ToleranceRelErr"})):
+        raise ValueError("Tolerance, ToleranceRelErr cannot be passed as additional parameters: set atol or rtol instead")
+    if atol < 0.0:
+        raise ValueError("Absolute tolerance must be nonnegative")
+    if rtol < 0.0:
+        raise ValueError("Relative tolerance must be nonnegative")
+
+    use_relative = rtol != Property.EMPTY_DBL
+    use_absolute = atol != Property.EMPTY_DBL
 
-    if rtol == Property.EMPTY_DBL and atol == Property.EMPTY_DBL:
-        raise ValueError("Specify rtol or atol")
+    # perform the required comparisons
 
-    if rtol > Property.EMPTY_DBL:
-        result, message = CompareWorkspaces(Workspace1, Workspace2, Tolerance=rtol, ToleranceRelErr=True, **kwargs)
+    # if neither set, do simple absolute comparison
+    if not use_absolute and not use_relative:
+        res, message = CompareWorkspaces(Workspace1, Workspace2, Tolerance=DEFAULT_ABS_TOLERANCE, ToleranceRelErr=False, **kwargs)
         msg_dict = message.toDict()
-        if not result:
-            msg = ", ".join(msg_dict["Message"]) + f", Workspaces are not within relative tolerance ({rtol})"
+        if res:
+            return
+        else:
+            msg = ", ".join(msg_dict["Message"]) + f", Workspaces are not within default tolerance ({DEFAULT_ABS_TOLERANCE})"
             raise AssertionError(msg)
 
-    if atol > Property.EMPTY_DBL:
-        result, message = CompareWorkspaces(Workspace1, Workspace2, Tolerance=atol, ToleranceRelErr=False, **kwargs)
-        msg_dict = message.toDict()
-        if not result:
-            msg = ", ".join(msg_dict["Message"]) + f", Workspaces are not within absolute tolerance ({atol})"
+    # if rtol set, perform relative comparison
+    if use_relative:
+        if "tes" in Workspace1.name():
+            print("USE RELATIVE")
+        rel_res, message = CompareWorkspaces(Workspace1, Workspace2, Tolerance=rtol, ToleranceRelErr=True, **kwargs)
+        rel_msg = message.toDict()
+        if "tes" in Workspace1.name():
+            print(f"REL RES {rel_res} - {rel_msg}")
```

Comment:
The logic of collecting the messages could be simplified if you added and else to both `CompareWorkspaces` call like
```python
else:
    rel_res, rel_msg = True, {}
```
then the code below could check `rel_res` and `abs_res`